<a href="https://colab.research.google.com/github/matheusrebe/intro_python/blob/main/exercicios_aula/aula_1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

1- Se você fizer uma corrida de 10 quilômetros em 43 minutos e 30 segundos, qual será seu tempo médio por milha? Qual é a sua velocidade média em milhas por hora? (Dica: há 1,61 quilômetros em uma milha). 

Como 1 km = 1,61 milha, podemos fazer a seguinte conta

In [ ]:
43.5/(10/1.61) #O termo entre parênteses é a conversão de quilometros para milhas

7.003500000000001

A velocidade média será:

In [ ]:
(10/1.61)*60/43.5 # 43.5/60 foi reescrito da forma final já para obter o valor correto

8.567144998929106

A velocidade média será de 8.567144998929106 m/s

2- Desde sua varanda você escuta o som do primeiro fogo artificial do reveillon 3 segundos depois de ver a luz, qual a distância? (o som tem velocidade 343 m/s e a luz $3\times 10^8$ m/s).


Como a velocidade da luz é gigantesca comparada a do som e levando em consideração que a visão humana tem uma visibilidade de até 10 km na nossa atmosfera, podemos considerar que o tempo que leva para visualizar o raio é 0 e basicamente o que será calculado é a distância que o som percorre em 3 s.

In [ ]:
v=343 #m/s
t=3 #s

print("A distância é de ",v*t, "m")

A distância é de  1029 m


A distância é de  1029 m

3- Ache os zeros da função

 y = $3x^2-4x -10 $


A fórmula de Bashkara é x=$\frac{-b \pm \sqrt(b^2-4ac))}{2a}$

O termo entre raiz é negativo, então teremos um número complexo w+iv

calculando então w e v:

In [ ]:
import math

a=3
b=-4
c=-10

w=(-b+math.sqrt(b**2-4*a*c))/(2*a)
v=(-b-math.sqrt(b**2-4*a*c))/(2*a)

print("As raízes valem ", '{:}'w, "e", v)

As raízes valem  2.610317298281767 e -1.2769839649484336


As raízes valem  2.610317298281767 e -1.2769839649484336

4- Se, ao meio dia, a sombra de um poste de 5 m de altura tem apenas 50 cm de comprimento no chão, qual o ângulo zenital do sol? 

Por semelhança de triangulos temos que o ângulo zenital do sol é o mesmo do formaldo pelo cateto da projeção da sombra e da altura do poste. Desse modo para obter o seno precisamos calculara a hipotenusa por pitágoras.

In [ ]:
import math

b=5 #m
c=0.5 #50 cm

a=math.sqrt(pow(b,2)+pow(c,2))

print("A hipotenusa vale ", a, "m")

#Agora então queremos o arcseno disso.

zenital=math.degrees(math.asin(0.5/a))

print("O ângulo zenital é", zenital, "graus")

A hipotenusa vale  5.024937810560445 m
O ângulo zenital é 5.710593137499642 graus


Portanto o ângulo zenital é 0.09966865249116202 rad
O ângulo zenital é 5.710593137499642 graus

In [ ]:
import math

help('math.asin') #Como o ângulo está em radianos o feito anterior está dimencionalmente ok!

Help on built-in function asin in math:

math.asin = asin(x, /)
    Return the arc sine (measured in radians) of x.



5-Calcule o seu IMC: IMC = M / $A^2$ (com a massa em Kg e a altura em metros). Um valor saudável estara --em geral-- entre 20-25. Um bebê de 6 meses "gorducho" tem 70 cm de "comprimento" e 11 kg de massa, qual o IMC dele? 

In [ ]:
#Calculando o IMC

i=11/pow(0.7,2) #0.7 é 70 cm em metros

print("O IMC deu ", i)

O IMC deu  22.44897959183674


O IMC deu  22.44897959183674, logo ele está na faixa saudável

6- Calcule a velocidade final de um objeto em queda livre a partir de 3 metros de altura (sem resistencia do ar). Calcule o tempo que esse objeto demora para cair. 

In [ ]:
#A equação da queda livre que relaciona as velocidades é dada por v=v0+gt, adotando o refrencial no sentido da queda. Podemos inclusive reduzi-la a v=gt
#Relacionando a posição sob mesmo referencial temos S=(gt^2)/2

#Primeiro achando o tempo que leva a queda.
import math

g=9.8 #m/s^2
S=3 #m

t=math.sqrt((2*S)/g)

#Usando isso para a equação das velocidades

v=g*t

print("A velocidade final é ", v)
print("O tempo de queda é ", t)

A velocidade final é  7.6681158050723255
O tempo de queda é  0.7824607964359516


A velocidade final é  7.6681158050723255 m/s

O tempo de queda é  0.7824607964359516 s